ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/tasks/Solaris.yml

```diff
@@ -15,26 +15,67 @@
   tags:
     - build_tools
 
-- name: Fetch CSWpkgutil.pkg
+- name: Fetch CSWpkgutil.pkg (x64)
   copy:
     src: files/CSWpkgutil.pkg
     dest: /tmp/CSWpkgutil.pkg
     force: yes
-  when: pkgutil_installed.rc != 0
+  when:
+    - pkgutil_installed.rc != 0
+    - ansible_architecture == "i386"
   tags: build_tools
 
-- name: Install pkgutil
+- name: Install pkgutil (x64)
   shell: |
     yes | pkgadd -d /tmp/CSWpkgutil.pkg CSWpkgutil
     /opt/csw/bin/pkgutil -U
-  when: pkgutil_installed.rc != 0
+  when:
+    - pkgutil_installed.rc != 0
+    - ansible_architecture == "i386"
+  tags: build_tools
+
+# The ansible copy command does not work at this point because it can't find source in the default shell
```

Comment:
Can you clarify this? What is it that's trying to use `source` instead of `.`?

Reply:
So we use the ansible copy module on x64 (as seen [here](https://github.com/adoptium/infrastructure/pull/2204/files#diff-5a3d2646edcdff57d8dcd180e791e61aa1c56e9a66bf853127032828f635b30cR18-R26)), but on Sparc the default shell that's installed doesn't seem to have `source` as a built-in command)

e.g

```bash
which source
no source in /usr/sbin /usr/bin /opt/csw/bin/
```

Whereas on my system:

```bash
which source
source: shell built-in command
```

Reply:
Yeah you'd normally use `.` in place of source which is why I'm wondering where `source` is being invoked from. It would be disappointing if that was sometihng deep within ansible that we can't control.